grails-app/controllers/org/pih/warehouse/api/CombinedShipmentItemApiController.groovy

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2012 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse.api
+
+import grails.converters.JSON
+import org.codehaus.groovy.grails.web.json.JSONObject
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.order.OrderItem
+import org.pih.warehouse.product.Product
+import org.pih.warehouse.shipping.Shipment
+import org.pih.warehouse.shipping.ShipmentItem
+
+class CombinedShipmentItemApiController {
+
+    def getOrderOptions = {
+        def vendor = Location.get(params.vendor)
+        def destination = Location.get(params.destination)
+        List<Order> orders = Order.findAllByOriginAndDestination(vendor, destination)
+        render([data: orders.findAll{ it.orderItems.any { item -> item.getQuantityRemainingToShip() > 0 } }.collect {
+            [
+                value: it.id,
+                label: it.orderNumber
+            ]
+        }] as JSON)
+    }
+
+    def findOrderItems = {
+        List<Order> orders
+        if (params.orderIds) {
+            orders = Order.findAllByIdInList(params.orderIds)
+        } else {
+            def vendor = Location.get(params.vendor)
+            def destination = Location.get(params.destination)
+            orders = Order.findAllByOriginAndDestination(vendor, destination)
```

Comment:
This probably needs to of PurchaseOrder type too. Just add a method to the service.